docs/userguide/datafiles.rst

```diff
@@ -407,49 +413,6 @@ which enables the ``data`` directory to be identified, and then, we separately s
 files for the root package ``mypkg``, and the namespace package ``data`` under the package
 ``mypkg``.
 
-With ``include_package_data`` the configuration is simpler: you simply need to enable
```

Comment:
Still, I don't think this part should be removed.

Reply:
I'm sorry I am taking so long to provide a good set of reproducers. I finally did a tally and came up with 48 cases. I suspect that it is going to end up that some of the config that the comments say is optional is not optional in my situation. After I have iterated through all the cases I'll give you a summary, a handful of the most instructive reproducers and hopefully a smaller edit of this section of the document, plus mentioning the `Manifest.in` option.

Reply:
@abravalheri After testing many of the configurations I am actually back to the original position, that this configuration does not permit you to include non-Python files in a distribution. I have only checked pyproject.toml configurations, so I do not know if the issue is specific to that code-path. I am including three reproducers each of which is using the configuration from the current documentation virtually verbatim, and that do not have any of the generated files in them. Two of them use a src parent directory, and one of them matches my own setup without a src parent directory.

[find-inc-yes-src-ns.tar.gz](https://github.com/pypa/setuptools/files/14299252/find-inc-yes-src-ns.tar.gz)
[find-inc-yes-src.tar.gz](https://github.com/pypa/setuptools/files/14299253/find-inc-yes-src.tar.gz)
[find-inc-yes-dot.tar.gz](https://github.com/pypa/setuptools/files/14299254/find-inc-yes-dot.tar.gz)

```
tar tzf find-inc-yes-dot.tar.gz
python -m build
zipinfo dist/find_where-0.0.dev0-py3-none-any.whl
```
substitute the other filenames as appropriate in the tar command.

I tried adding the package-data configuration described above the section I suggest removing to show things working, with the result that the namespace-level HTML file is included but only the Python is included in the package. At this point I remembered that I had previously avoided including dashes in package directories because they are interpreted as subtraction in Python. The same configuration with the directory renamed to have an underscore rather than a dash results in all of the HTML files being included. The same directory name with the package-data configuration commented out again results in no HTML files being included. I do not know if we should also include a note about dashes in package directory names not supporting data files, or if there's a different way to configure the package data in that situation, or whether there's already a more central piece of documentation saying that it's a bad idea and will cause random things to break even if it sometimes works.

[find_where-inc-yes-src.tar.gz](https://github.com/pypa/setuptools/files/14300079/find_where-inc-yes-src.tar.gz)
[find_where-pd-inc-yes-src.tar.gz](https://github.com/pypa/setuptools/files/14300080/find_where-pd-inc-yes-src.tar.gz)
[find-where-pd-inc-yes-src.tar.gz](https://github.com/pypa/setuptools/files/14300081/find-where-pd-inc-yes-src.tar.gz)

Reply:
Hi @thrasymache thank you very much for the in-depth investigation.
I just want to double check with you a couple of point before delving into the reproducers you provided:

- When you say "this configuration does not permit you to include non-Python files in a distribution", I am assuming you are referring to `include-package-data` (whose default value is `True` when it comes from `pyproject.toml`). Please let me know if you are referring to the `package-data` configuration field instead.
- I am assuming, you considered this passage on the [docs](https://setuptools.pypa.io/en/latest/userguide/datafiles.html#include-package-data)(that I have [previously highlighted](https://github.com/pypa/setuptools/pull/4193#issuecomment-1913612377) in the conversation):
  > [...] you can simply use the `include_package_data` keyword [...]
  > [...] files will be automatically installed with your package, **provided**:
  > - These files are included via the [MANIFEST.in] 
      (https://setuptools.pypa.io/en/latest/userguide/miscellaneous.html#using-manifest-in) file, [...]
     > - OR, they are being tracked by a revision control system such as Git, Mercurial or SVN, and you have configured an appropriate plugin such as [setuptools-scm](https://pypi.org/project/setuptools-scm) or [setuptools-svn](https://pypi.org/project/setuptools-svn) [...]

    If that is not the case please let me know. (Also if none of these conditions is met, I agree 100% that the outcome is that the files will be not included at the end, but that scenario is described by the docs).

Reply:
Now we're getting to the heart of the matter. I think I can save you the time of delving into the reproducers: Yes, I am using the `include-package-data` configuration, and none of the reproducers use either `MANIFEST.in` nor a version control plugin, but there is still a glaring problem with the documentation as it stands: the passage that you are highlighting is in the `include_package_data` section, and the edit that I started suggesting is in the `Subdirectory for Data Files` section. Now that I have been looking at this document for a few weeks I can see how it's trying to work, but also how easy it is for someone looking at it the first time to read it the way that I did last month.

There are 4 sections before the Summary and Addenda. The first two specify independent ways to include data files which can also be used together. The third section (`exclude_package_data`) only makes sense if being used with one or both of the first two, but that is also clear from its content. The fourth `Subdirectory for Data Files` section, is actually describing the same configurations as the first two, but it is incomplete, in that it doesn't mention that `include-package-data` requires the specific files to be specified elsewhere. This is something that is clear in the first section and in the Summary, though it isn't clear in the fourth section that you also need to read anything else. Specifically, it says:

> With include_package_data the configuration is simpler: you simply need to enable scanning of namespace packages in the src directory and the rest is handled by Setuptools.

This gives the impression that the below configuration is all you need, since specifying the packages elsewhere in the same file is simpler than creating a new file to do it or installing a plugin. My guess is that the author of this section wasn't aware that he was depending upon the plugin for this configuration to work.

Anyway, I agree with the `MANIFEST.in` the data files would be included by the config that I suggested removing (although it might be worth noting that in a `pyproject.toml` and the right layout you can get away with omitting it). The conservative edit would be to simply mention the necessity of either the `MANIFEST.in` or the VCS plugin here as it is elsewhere on the page. On the other hand, if you're up for it, I still think it's confusing to have this section here discussing two of the three configuration options in a place in the document where you would expect a fourth configuration option. I think it would make more sense if this section was at least below the Summary, and referred to the sections that give details on the two configuration options in addition or instead of explicit configs.

Reply:
Thank you very much @thrasymache. That is a very good summary!

I am proposing an alternative in #4230. Please feel free to comment.